addons/persistence/mongodb-persistence-addon/src/main/java/org/kie/kogito/mongodb/DocumentMarshallingStrategy.java

```diff
@@ -19,76 +19,67 @@
 import java.io.ObjectInputStream;
 import java.io.ObjectOutputStream;
 
-import org.kie.api.marshalling.ObjectMarshallingStrategy;
-import org.kie.kogito.mongodb.utils.CommonUtils;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.JsonMappingException;
+import org.kie.api.marshalling.ObjectMarshallingStrategy;
+import org.kie.kogito.mongodb.utils.CommonUtils;
 
 public class DocumentMarshallingStrategy implements ObjectMarshallingStrategy {
-	
-    private static final Logger LOGGER = LoggerFactory.getLogger(DocumentMarshallingStrategy.class);
-    
-	public String marshalToJson(Object object) {
 
-		String json = null;
-		try {
-			json = CommonUtils.getObjectMapper().writeValueAsString(object);
-		} catch (JsonProcessingException e) {
-			 LOGGER.error("Error while writing object as json", e);
-		}
-		return json;
+    @Override
+    public String marshalToJson(Object object) {
+        try {
+            return CommonUtils.getObjectMapper().writeValueAsString(object);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("JsonProcessingException writing object as json : " + e.getMessage(), e);
+        }
+    }
 
-	}
+    @Override
+    public Object unmarshalFromJson(String dataType, String json) {
+        try {
+            Class<?> loadClass = Thread.currentThread().getContextClassLoader().loadClass(dataType);
+            return CommonUtils.getObjectMapper().readValue(json, loadClass);
+        } catch (ClassNotFoundException e) {
+            throw new IllegalArgumentException("ClassNotFoundException while reading object from json : " + e.getMessage(), e);
+        } catch (JsonMappingException e) {
+            throw new IllegalArgumentException("JsonMappingException while reading object from json : " + e.getMessage(), e);
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException("JsonProcessingException while reading object fom json : " + e.getMessage(), e);
+        }
```

Comment:
Same here, unmarshalling a JSON into an arbitrary object should throw a catched exception but here as you have more things that might fail I'd recommend you to create a dedicated exception that extends Exception. e.g UnmashalException.
Do not repeat the same catch for different exception types. You can catch a superclass or use multi-catch.
```suggestion
        } catch (ClassNotFoundException | JsonMappingException | JsonProcessingException e) {
            throw new UnmarshalException("Unable to unmarshal from json", e);
        }
```